src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2Extensions.cs

```diff
@@ -203,5 +203,31 @@ public static ArraySegment<byte> GetArraySegment(this MemoryPoolIterator2 start,
             start.CopyTo(array, 0, length, out length);
             return new ArraySegment<byte>(array, 0, length);
         }
+
+        public static bool StartsWithIgnoreCase(this MemoryPoolIterator2 start, string prefix, out MemoryPoolIterator2 end, out bool caseMatches)
+        {
+            end = start;
+            caseMatches = true;
+
+            var i = 0;
+            for (i = 0; i < prefix.Length; i++)
+            {
+                var c = end.Take();
+
+                if (c != prefix[i])
+                {
+                    if (char.ToLowerInvariant((char)c) == char.ToLowerInvariant(prefix[i]))
```

Comment:
@halter73 char.ToLowerInvariant already optimizes the call when it's ASCII. It ends up here:

http://referencesource.microsoft.com/#mscorlib/system/globalization/textinfo.cs,145c4c32b3b4bace